### PR TITLE
Rename snapshot -> backup

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

- Rename all instances of "snapshot" to "backup"
- Adds a redirect from /hassio/snapshots to /hassio/backups

Related:
<https://github.com/home-assistant/core/pull/51629>
<https://github.com/home-assistant/supervisor/pull/2940>
<https://github.com/home-assistant/my.home-assistant.io/pull/95>
<https://github.com/home-assistant/home-assistant.io/pull/18129>
<https://github.com/home-assistant/developers.home-assistant/pull/963>
<https://github.com/home-assistant/cli/pull/301>
<https://github.com/home-assistant/addons/pull/2088>

Prior to merge these keys needs to be renamed/deleted from localize:

old key | action | new key
-- | -- | --
`supervisor.snapshot.delete_snapshot_confirm` | rename | `supervisor.backup.delete_backup_confirm`
`supervisor.snapshot.selected` | rename | `supervisor.backup.selected`
`supervisor.snapshot.failed_to_delete` | rename | `supervisor.backup.failed_to_delete`
`supervisor.snapshot.create` | rename | `supervisor.backup.create`
`supervisor.snapshot.created` | rename | `supervisor.backup.created`
`supervisor.snapshot.select_type` | rename | `supervisor.backup.select_type`
`supervisor.snapshot.addons` | rename | `supervisor.backup.addons`
`supervisor.snapshot.folders` | rename | `supervisor.backup.folders`
`supervisor.snapshot.password_protection` | rename | `supervisor.backup.password_protection`
`supervisor.snapshot.enter_password` | rename | `supervisor.backup.enter_password`
`supervisor.snapshot.passwords_not_matching` | rename | `supervisor.backup.passwords_not_matching`
`supervisor.snapshot.select_type` | rename | `supervisor.backup.select_type`
`supervisor.panel.snapshots` | delete | 
`page-onboarding.restore.description` | delete | 
`supervisor.snapshot.description` | delete | 
`supervisor.snapshot.available_snapshots` | delete | 
`supervisor.snapshot.no_snapshots` | delete | 
`supervisor.snapshot.create_blocked_not_running` | delete | 
`supervisor.snapshot.delete_selected` | delete | 
`supervisor.snapshot.delete_snapshot_title` | delete | 
`supervisor.snapshot.delete_snapshot_text` | delete | 
`supervisor.snapshot.could_not_create` | delete | 
`supervisor.snapshot.upload_snapshot` | delete | 
`supervisor.snapshot.create_snapshot` | delete | 
`supervisor.snapshot.name` | delete | 
`supervisor.snapshot.type` | delete | 
`supervisor.snapshot.security` | delete | 
`supervisor.snapshot.full_snapshot` | delete | 
`supervisor.snapshot.password` | delete | 
`supervisor.snapshot.confirm_password` | delete | 
`supervisor.snapshot.password_protected` | delete | 
`supervisor.snapshot.folder.homeassistant` | delete | 
`supervisor.snapshot.folder.ssl` | delete | 
`supervisor.snapshot.folder.share` | delete | 
`supervisor.snapshot.folder.homeassistant` | delete | 
`supervisor.snapshot.folder.addons/local` | delete | 
`supervisor.dialog.update.snapshot` | delete | 
`supervisor.dialog.update.create_snapshot` | delete | 
`supervisor.dialog.update.snapshotting` | delete | 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/hassio-panel-router.ts [hunk 1]
```diff
@@ -23,6 +24,7 @@ class HassioPanelRouter extends HassRouterPage {
   @property({ type: Boolean }) public narrow!: boolean;
 
   protected routerOptions: RouterOptions = {
+    initialLoad: () => this._redirects(),
```
Comment (balloob) [inline hassio/src/hassio-panel-router.ts:27]: Use the `beforeRender` option instead. 

```suggestion
    beforeRender: (page: string) => page === "snapshots" ? "backups" : undefined,
```

File: hassio/src/hassio-panel-router.ts [hunk 2]
```diff
@@ -45,6 +50,12 @@ class HassioPanelRouter extends HassRouterPage {
     el.route = this.route;
     el.narrow = this.narrow;
   }
+
+  private async _redirects() {
+    if (this.route.path === "/snapshots") {
+      navigate("/hassio/backups", { replace: true });
+    }
+  }
```
Comment (balloob) [inline hassio/src/hassio-panel-router.ts:58]:
```suggestion
```

File: hassio/src/hassio-router.ts [hunk 3]
```diff
@@ -23,14 +24,15 @@ class HassioRouter extends HassRouterPage {
   protected routerOptions: RouterOptions = {
     // Hass.io has a page with tabs, so we route all non-matching routes to it.
     defaultPage: "dashboard",
-    initialLoad: () => this._redirectIngress(),
+    initialLoad: () => this._redirects(),
```
Comment (balloob) [inline hassio/src/hassio-router.ts:27]: Same.

File: src/translations/en.json [hunk 4]
```diff
@@ -3899,43 +3899,32 @@
         "ram_usage": "Core RAM Usage"
       }
     },
-    "snapshot": {
```
Comment (balloob) [inline src/translations/en.json:3902]: We should rename these keys in Localize before we merge this PR so we don't lose translations.

File: hassio/src/components/hassio-upload-snapshot.ts [hunk 5]
```diff
@@ -50,7 +50,7 @@ export class HassioUploadSnapshot extends LitElement {
 
     if (file.size > MAX_FILE_SIZE) {
       showAlertDialog(this, {
-        title: "Snapshot file is too big",
+        title: "Bakcup file is too big",
```
Comment (bramkragten) [inline hassio/src/components/hassio-upload-snapshot.ts:53]:
```suggestion
        title: "Backup file is too big",
```

File: src/translations/en.json (hunk 4 above)
Comment (ludeeus) [reply src/translations/en.json:3902]: No all, those containing "snapshot" should be removed.

File: hassio/src/hassio-panel-router.ts (hunk 1 above)
Comment (bramkragten) [reply hassio/src/hassio-panel-router.ts:27]: I thought the `routeOption` could just be a string to redirect?

File: hassio/src/hassio-panel-router.ts [hunk 6]
```diff
@@ -33,6 +35,9 @@ class HassioPanelRouter extends HassRouterPage {
       snapshots: {
         tag: "hassio-snapshots",
       },
```
Comment (bramkragten) [inline hassio/src/hassio-panel-router.ts:37]:
```suggestion
      snapshots: "backups",
```

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**